src/core/server/utils/workspace.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchDashboardsRequest, ensureRawRequest } from '../http/router';
+
+export interface WorkspaceState {
+  requestWorkspaceId?: string;
+}
+
+/**
+ * This function will be used as a proxy
+ * because `ensureRequest` is only importable from core module.
+ *
+ * @param workspaceId string
+ * @returns void
+ */
+export const updateWorkspaceState = (
+  request: OpenSearchDashboardsRequest,
+  payload: Partial<WorkspaceState>
```

Comment:
looks like workspacestate only has one field which is optional, is it necessary to specify partial?

Reply:
I think yes because actually the function just spread all the property into the request.app, which means user can pass in partial of the `WorkspaceState` to update the state.